redfish-core/include/utils/chassis_utils.hpp

```diff
@@ -61,5 +61,258 @@ void getValidChassisID(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                return;
+            }
+
+            std::vector<std::string> sortedAssemblyList = *assemblyList;
+            std::sort(sortedAssemblyList.begin(), sortedAssemblyList.end());
+            checkAssemblyInterface(aResp, chassisPath, sortedAssemblyList,
+                                   std::move(callback));
+            return;
+        },
+        "xyz.openbmc_project.ObjectMapper", assemblyPath,
+        "org.freedesktop.DBus.Properties", "Get",
+        "xyz.openbmc_project.Association", "endpoints");
+}
+
+template <typename Callback>
+inline void checkForAssemblyAssociations(
+    const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+    const std::string& chassisPath, const std::string& service,
+    Callback&& callback)
+{
+    BMCWEB_LOG_DEBUG << "Check for assembly association";
+
+    using associationList =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+
+    crow::connections::systemBus->async_method_call(
+        [aResp, chassisPath, callback{std::move(callback)}](
+            const boost::system::error_code ec,
+            const std::variant<associationList>& associations) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const associationList* value =
+                std::get_if<associationList>(&associations);
+            if (value == nullptr)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            bool isAssmeblyAssociation = false;
+            for (const auto& listOfAssociations : *value)
+            {
+                if (std::get<0>(listOfAssociations) != "assembly")
+                {
+                    // implies this is not an assembly
+                    // association
+                    continue;
+                }
+
+                isAssmeblyAssociation = true;
+                break;
+            }
+
+            if (isAssmeblyAssociation)
+            {
+                getAssemblyEndpoints(aResp, chassisPath, std::move(callback));
+            }
+        },
+        service, chassisPath, "org.freedesktop.DBus.Properties", "Get",
+        "xyz.openbmc_project.Association.Definitions", "Associations");
+}
+
+template <typename Callback>
+inline void checkAssociation(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const std::string& chassisPath,
+                             Callback&& callback)
+{
+    BMCWEB_LOG_DEBUG << "Check chassis for association";
+
+    std::string chassis =
+        sdbusplus::message::object_path(chassisPath).filename();
+    if (chassis.empty())
+    {
+        BMCWEB_LOG_ERROR << "Failed to find / in Chassis path";
+        messages::internalError(aResp->res);
+        return;
+    }
```

Comment:
I don't see the need for the above check since we already know that chassis is not empty, and is "chassis" at this point. If it is not then this method wouldn't get called.

Reply:
we are checking this in case object_path unable to retrieve chassis data

Reply:
Isn't that already checked at line number 283 for the same path?

Reply:
Isn't that already checked at line number 283 for the same path?